src/bp/core/config/config-writer.ts

```diff
@@ -1,20 +1,85 @@
+import { BotTemplate, Logger } from 'botpress/sdk'
+import { Bot } from 'core/misc/interfaces'
+import { CMSService } from 'core/services/cms'
+import { FlowService } from 'core/services/dialog/flow/service'
+import { ModuleResourceLoader } from 'core/services/module/resources-loader'
+import { TYPES } from 'core/types'
 import fse from 'fs-extra'
-import { injectable } from 'inversify'
+import { inject, injectable } from 'inversify'
+import defaultJsonBuilder from 'json-schema-defaults'
+import _ from 'lodash'
 import path from 'path'
 import { VError } from 'verror'
 
 import { BotConfig } from './bot.config'
 
 @injectable()
 export class BotConfigWriter {
-  async writeToFile(config: BotConfig) {
+  private BOT_CONFIG_FILENAME = 'bot.config.json'
+  private BOT_SCHEMA_FILENAME = 'bot.config.schema.json'
+
+  constructor(
+    @inject(TYPES.Logger) private logger: Logger,
+    @inject(TYPES.CMSService) private cms: CMSService,
+    @inject(TYPES.FlowService) private flowService: FlowService
```

Comment:
I don't think it makes sense to have the config writer depends on CMS and Flow services.  The methods required on those two services should probably be extracted or else we can probably change the logic here (for example by passing the CMS elements as parameters)